src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,55 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private static final int INITIAL_LENGTH = 10;
+
+    private T[] elementData;
+    private int size;
 
+    public MateStack() {
+        elementData = (T[]) new Object[INITIAL_LENGTH];
+    }
+
+    public void push(T value) {
+        if (!checkCapacity()) {
+            resize();
+        }
+        elementData[size] = value;
+        size++;
     }
 
     public T peek() {
-        return null;
+        if (size() == 0) {
+            throw new EmptyStackException();
+        }
+        return elementData[size - 1];
     }
 
     public T pop() {
-        return null;
+        if (size() == 0) {
+            throw new EmptyStackException();
+        }
+        size--;
+        return elementData[size];
```

Comment:
this object will still remain in the array. Try to use System.arraycopy for this

Reply:
done